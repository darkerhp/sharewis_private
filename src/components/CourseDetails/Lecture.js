import React from 'react';
import ReactNative from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

import BaseStyles from '../../baseStyles';
import * as ApiConstants from '../../constants/Api';
import * as LectureUtils from '../../utils/lecture';
import Duration from '../Duration';

const { PropTypes } = React;
const { Platform, StyleSheet, Text, TouchableOpacity, View } = ReactNative;

const lectureRowHeight = 48;
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
    flexDirection: 'row',
    borderColor: BaseStyles.borderColor,
    borderBottomWidth: 1,
  },
  durationStyle: {
    ...Platform.select({
      android: {
        color: BaseStyles.textColor,
        width: 30,  // or seconds will be trimmed in android
      },
    }),
  },
  lectureNoTextWrapper: {
    flex: 1,
    height: lectureRowHeight,
    justifyContent: 'center',
    alignItems: 'center',
    borderColor: BaseStyles.borderColor,
    borderRightWidth: 1,
  },
  lectureNoTextWrapperCompleted: { // FIXME lectureNoTextWrapperと共通化できる？
    flex: 1,
    height: lectureRowHeight,
    justifyContent: 'center',
    alignItems: 'center',
    borderColor: BaseStyles.backgroundColor,
    borderRightWidth: 1,
    backgroundColor: BaseStyles.backgroundColor,
  },
  lectureNoText: {
    fontSize: 14,
    color: BaseStyles.textColor,
    fontWeight: '600',
  },
  lectureInfoWrapper: {
    flex: 1,
    height: lectureRowHeight,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  lectureIconWrapper: {
    flex: 1,
    justifyContent: 'center',
  },
  lectureIcon: {
    color: BaseStyles.textColor,
  },
  durationWrapper: {
    flex: 1,
  },
  lectureTitleTextWrapper: {
    flex: 5,
    height: lectureRowHeight,
    justifyContent: 'center',
    alignItems: 'flex-start',
  },
  lectureTitleText: {
    fontSize: 12,
    color: BaseStyles.textColor,
    fontWeight: 'bold',
  },
  actionIconWrapper: {
    flex: 1,
    height: lectureRowHeight,
    justifyContent: 'center',
    alignItems: 'center',
  },
  actionIcon: {
    color: '#757575',
    fontSize: 32,
  },
});


const renderDownloadAction = () =>
  <TouchableOpacity style={styles.actionIconWrapper}>
    <Icon
      // name={'delete'}
      name={'cloud-download'}
      style={styles.actionIcon}
    />
  </TouchableOpacity>;

const Lecture = ({ lectures, currentLecture, handlePressLecture }) => {
  const isAccessibleLecture = currentLecture.type === ApiConstants.LECTURE_TYPE_VIDEO;
  return (
    <View style={[styles.container, (!isAccessibleLecture ? { backgroundColor: 'lightgray' } : {})]}>
      <View
        style={(currentLecture.status === 'finished')
                ? styles.lectureNoTextWrapperCompleted
                : styles.lectureNoTextWrapper}
      >
        <Text style={styles.lectureNoText}>{currentLecture.order}</Text>
      </View>

      <View style={styles.lectureInfoWrapper}>
        <View style={styles.lectureIconWrapper}>
          <Icon
            name={LectureUtils.getLectureIconName(currentLecture)}
            style={styles.lectureIcon}
          />
        </View>
        <Duration
          estimatedTime={currentLecture.estimatedTime}
          containerStyleId={styles.durationWrapper}
          durationStyleId={styles.durationStyle}
        />
      </View>

      <TouchableOpacity
        style={[styles.lectureTitleTextWrapper, (!isAccessibleLecture ? { flex: 6 } : {})]}
        onPress={() => handlePressLecture(currentLecture, lectures)}
        disabled={!isAccessibleLecture}
      >
        <Text style={styles.lectureTitleText}>{currentLecture.title}</Text>
      </TouchableOpacity>

      {isAccessibleLecture && renderDownloadAction()}
    </View>
  );
};


Lecture.propTypes = {
  lectures: PropTypes.arrayOf(PropTypes.shape({})),
  currentLecture: PropTypes.shape({
    /* eslint-disable react/no-unused-prop-types */
    order: PropTypes.number,
    title: PropTypes.string,
    kind: PropTypes.string,
    estimatedTime: PropTypes.number,
    status: PropTypes.string,
    type: PropTypes.string,
    /* eslint-enable react/no-unused-prop-types */
  }).isRequired,
  handlePressLecture: PropTypes.func.isRequired,
};


export default Lecture;
