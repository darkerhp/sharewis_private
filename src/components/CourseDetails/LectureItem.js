import React from 'react';
import ReactNative from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import { AnimatedCircularProgress } from 'react-native-circular-progress';

import BaseStyles from '../../baseStyles';
import * as LectureUtils from '../../utils/lecture';
import { LECTURE_TYPE_VIDEO, LECTURE_STATUS_FINISHED } from '../../constants/Api';
import Duration from '../Duration';

const { Component, PropTypes } = React;
const { Platform, StyleSheet, Text, TouchableOpacity, View } = ReactNative;

const lectureRowHeight = 48;
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
    flexDirection: 'row',
    borderColor: BaseStyles.borderColor,
    borderBottomWidth: 1,
  },
  lectureNoTextWrapper: {
    flex: 1,
    height: lectureRowHeight,
    justifyContent: 'center',
    alignItems: 'center',
    borderColor: BaseStyles.borderColor,
    borderRightWidth: 1,
  },
  lectureNoTextWrapperCompleted: { // FIXME lectureNoTextWrapperと共通化できる？
    flex: 1,
    height: lectureRowHeight,
    justifyContent: 'center',
    alignItems: 'center',
    borderColor: BaseStyles.backgroundColor,
    borderRightWidth: 1,
    backgroundColor: BaseStyles.backgroundColor,
  },
  lectureNoText: {
    fontSize: 14,
    color: BaseStyles.textColor,
    fontWeight: '600',
  },
  lectureInfoWrapper: {
    flex: 1,
    height: lectureRowHeight,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  lectureIconWrapper: {
    flex: 1,
    justifyContent: 'center',
  },
  durationWrapper: {
    flex: 1,
  },
  durationStyle: {
    ...Platform.select({
      android: {
        color: BaseStyles.textColor,
        width: 30,  // or seconds will be trimmed in android
      },
    }),
  },
  lectureTitleTextWrapper: {
    flex: 5,
    height: lectureRowHeight,
    justifyContent: 'center',
    alignItems: 'flex-start',
  },
  lectureTitleText: {
    fontSize: 12,
    color: BaseStyles.textColor,
    fontWeight: 'bold',
  },
  actionIconWrapper: {
    flex: 1,
    height: lectureRowHeight,
    justifyContent: 'center',
    alignItems: 'center',
  },
  actionIcon: {
    color: '#757575',
    fontSize: 32,
  },
});

// TODO components化
const renderDownloadAction = (handlePressDownload, lecture) =>
  <TouchableOpacity
    style={styles.actionIconWrapper}
    onPress={() => handlePressDownload(lecture)}
  >
    {lecture.isDownloading
      ?
      <AnimatedCircularProgress
        size={30}
        width={3}
        fill={lecture.percentage}
        rotation={0}
        tintColor="#00e0ff"
        backgroundColor="#3d5875"
      />
      :
      <Icon
        name={lecture.hasVideoInDevice ? 'delete' : 'cloud-download'}
        style={styles.actionIcon}
      />
    }
  </TouchableOpacity>;

class LectureItem extends Component {
  static propTypes = {
    currentLecture: PropTypes.shape({}).isRequired,
    handlePressLecture: PropTypes.func.isRequired,
    handlePressDownload: PropTypes.func.isRequired,
    fetchDownloadStatus: PropTypes.func.isRequired,
    courseId: PropTypes.number.isRequired,
  };

  componentDidMount() {
    const { courseId, currentLecture, fetchDownloadStatus } = this.props;
    fetchDownloadStatus(courseId, currentLecture.id);
  }

  render() {
    const { currentLecture, handlePressLecture, handlePressDownload } = this.props;
    const isAccessibleLecture = currentLecture.type === LECTURE_TYPE_VIDEO;
    return (
      <View style={[styles.container, (!isAccessibleLecture ? { backgroundColor: 'lightgray' } : {})]}>
        <View
          style={currentLecture.status === LECTURE_STATUS_FINISHED
            ? styles.lectureNoTextWrapperCompleted
            : styles.lectureNoTextWrapper}
        >
          <Text style={styles.lectureNoText}>{currentLecture.order}</Text>
        </View>

        <View style={styles.lectureInfoWrapper}>
          <View style={styles.lectureIconWrapper}>
            <Icon name={LectureUtils.getLectureIconName(currentLecture)} />
          </View>
          <Duration
            estimatedTime={currentLecture.estimatedTime}
            containerStyleId={styles.durationWrapper}
            durationStyleId={styles.durationStyle}
          />
        </View>

        <TouchableOpacity
          style={[styles.lectureTitleTextWrapper, (!isAccessibleLecture ? { flex: 6 } : {})]}
          onPress={() => handlePressLecture(currentLecture)}
          disabled={!isAccessibleLecture}
        >
          <Text style={styles.lectureTitleText}>{currentLecture.title}</Text>
        </TouchableOpacity>

        {isAccessibleLecture && renderDownloadAction(handlePressDownload, currentLecture)}
      </View>
    );
  }
}

export default LectureItem;
