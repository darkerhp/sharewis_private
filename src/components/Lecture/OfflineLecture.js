import React, { PropTypes } from 'react';
import ReactNative from 'react-native';
import I18n from 'react-native-i18n';

const { View, StyleSheet, Text } = ReactNative;


const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  contentText: {
    textAlign: 'center',
    lineHeight: 22,
    fontSize: 17,
    color: '#222',
  },
});


const OfflineLecture = ({ lectureContentStyleId }) =>
  (
    <View style={[lectureContentStyleId, styles.container]}>
      <View>
        <Text style={styles.contentText}>
          {I18n.t('offlineLecture')}
        </Text>
      </View>
    </View>
  );

OfflineLecture.propTypes = {
  lectureContentStyleId: PropTypes.number.isRequired,
};

export default OfflineLecture;
