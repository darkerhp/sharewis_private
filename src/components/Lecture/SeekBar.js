import React from 'react';
import ReactNative from 'react-native';

import Duration from '../Duration';

const { PropTypes } = React;
const { Platform, View, StyleSheet, Slider, Dimensions } = ReactNative;

const DEVICE_WIDTH = Dimensions.get('window').width;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'space-around',
    alignItems: 'center',
    flexDirection: 'row',
  },
  slider: {
    width: (DEVICE_WIDTH - 90) * 0.8,
    ...Platform.select({
      android: {
        backgroundColor: '#eee',
      },
    }),
  },
  timeTextContainer: {
    flex: 1,
    justifyContent: 'center',
    width: 60,
  },
  timeText: {
    color: '#bdbdbd',
    fontSize: 16,
    fontWeight: 'bold',
    opacity: 0.8,
    padding: 0,
    textAlign: 'center',
  },
});

const SeekBar = ({ currentTime, estimatedTime, onValueChange }) => {
  const timeEnd = estimatedTime - currentTime;
  return (
    <View style={styles.container}>
      <Duration
        estimatedTime={currentTime}
        format={'mm:ss'}
        containerStyleId={styles.timeTextContainer}
        durationStyleId={styles.timeText}
      />
      <Slider
        maximumValue={estimatedTime}
        value={currentTime}
        onValueChange={value => onValueChange(value)}
        style={styles.slider}
      />
      <Duration
        estimatedTime={timeEnd < 0 ? 0 : timeEnd}
        format={'mm:ss'}
        containerStyleId={styles.timeTextContainer}
        durationStyleId={styles.timeText}
        prefixText={'-'}
      />
    </View>
  );
};


SeekBar.propTypes = {
  currentTime: PropTypes.number.isRequired,
  estimatedTime: PropTypes.number.isRequired,
  onValueChange: PropTypes.func.isRequired,
};

export default SeekBar;
