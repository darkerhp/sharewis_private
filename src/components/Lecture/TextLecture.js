import React, { Component, PropTypes } from 'react';
import ReactNative from 'react-native';

import Lecture from '../../modules/models/Lecture';
import BaseStyles from '../../lib/baseStyles';

const {
  StyleSheet,
  Text,
  View,
  WebView,
} = ReactNative;

const styles = StyleSheet.create({
  webViewWrapper: { flex: 9 },
  webView: {},
  lectureTitleTextWrapper: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  lectureTitle: {
    fontSize: 17,
    color: '#e0e0e0',
    fontWeight: 'bold',
  },
});

const TextLecture = ({ currentLecture, lectureContentStyleId }) =>
  (
    <View style={[lectureContentStyleId, { marginTop: BaseStyles.navbarHeight }]}>
      <View style={styles.lectureTitleTextWrapper}>
        <Text style={styles.lectureTitle}>{currentLecture.title}</Text>
      </View>
      <View style={styles.webViewWrapper}>
        <WebView
          style={styles.webView}
          source={{ html: currentLecture.body }}
        />
      </View>
    </View>
  );

TextLecture.propTypes = {
  currentLecture: PropTypes.instanceOf(Lecture).isRequired,
  lectureContentStyleId: PropTypes.number.isRequired,
};

export default TextLecture;
