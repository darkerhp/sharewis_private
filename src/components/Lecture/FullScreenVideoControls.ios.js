// NOTE iOS版はreact-native-videoライブラリの不具合によりプログレスバーを表示しない
import React, { Component, PropTypes } from 'react';
import ReactNative from 'react-native';

import autobind from 'autobind-decorator';
import Button from 'react-native-button';
import Icon from 'react-native-vector-icons/MaterialIcons';

import SeekBar from './SeekBar';
import BaseStyles from '../../lib/baseStyles';

const { Platform, StyleSheet, TouchableOpacity, Text, View } = ReactNative;

const styles = StyleSheet.create({
  containerWrapper: {
    position: 'absolute',
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
  },
  container: {
    backgroundColor: 'rgba(0,0,0,0.3)',
    position: 'absolute',
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
  },
  headerWrapper: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
    flexDirection: 'row',
  },
  buttonWrapper: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
  },
  seekBarWrapper: {
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'flex-end',
    flexDirection: 'row',
    ...Platform.select({
      android: {
        paddingBottom: 10,
      },
    }),
  },
  playButton: {
    width: 78,
    height: 78,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'transparent',
  },
  playButtonIcon: {
    fontSize: 78,
    height: 78,
    color: 'white',
    opacity: 0.9,
  },
  buttonText: {
    fontSize: 18,
    color: 'white',
  },
  lectureTitleTextWrapper: {
    flex: 0.5,
    justifyContent: 'center',
    alignItems: 'flex-start',
    padding: 15,
  },
  lectureTitle: {
    fontSize: 20,
    color: 'white',
    fontWeight: 'bold',
    opacity: 0.9,
  },
  fullScreenExitButton: {
    width: 34,
    height: 34,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'transparent',
    ...Platform.select({
      ios: {
        paddingBottom: 5,
      },
      android: {
        paddingTop: 10,
      },
    }),
  },
  fullScreenExitButtonIcon: {
    fontSize: 34,
    height: 34,
    color: 'white',
    opacity: 0.9,
  },
});

class FullScreenVideoControls extends Component {
  static propTypes = {
    isLoadingThumbnail: PropTypes.bool.isRequired,
    isPaused: PropTypes.bool.isRequired,
    title: PropTypes.string.isRequired,
    //
    onPressFullScreen: PropTypes.func.isRequired,
    onPressPlay: PropTypes.func.isRequired,
  };

  state = {
    isShowControls: false,
    timeoutId: 0,
  };

  @autobind
  handlePressContainer() {
    const { isPaused } = this.props;
    // 再生中に画面がタップされたら5秒間コントロールを表示する
    if (!isPaused) {
      this.setState({ isShowControls: true });
      if (this.state.timeoutId) clearTimeout(this.state.timeoutId);
      const timeoutId = setTimeout(() => this.setState({ isShowControls: false }), 5000);
      this.setState({ timeoutId });
    }
  }

  @autobind
  handlePressFullScreenExit() {
    const { onPressFullScreen } = this.props;
    if (this.state.timeoutId) clearTimeout(this.state.timeoutId);
    onPressFullScreen();
  }

  render() {
    const {
      isLoadingThumbnail,
      isPaused,
      onPressPlay,
      title,
    } = this.props;

    return (
      <TouchableOpacity
        style={styles.containerWrapper}
        onPress={this.handlePressContainer}
        activeOpacity={1}
      >
        {(isPaused || this.state.isShowControls) &&
        <View style={styles.container}>
          <View style={styles.headerWrapper}>
            <View style={styles.lectureTitleTextWrapper}>
              <Text style={styles.lectureTitle}>{title}</Text>
            </View>
          </View>
          <View style={styles.buttonWrapper}>
            <Button
              containerStyle={[
                styles.playButton,
                isLoadingThumbnail && { backgroundColor: BaseStyles.disabledButtonColor },
              ]}
              style={styles.buttonText}
              onPress={() => onPressPlay()}
              disabled={isLoadingThumbnail}
            >
              <Icon
                name={isPaused ? 'play-arrow' : 'pause'}
                style={styles.playButtonIcon}
              />
            </Button>
          </View>
          <View style={styles.seekBarWrapper}>
            <Button
              containerStyle={styles.fullScreenExitButton}
              style={styles.buttonText}
              onPress={this.handlePressFullScreenExit}
            >
              <Icon
                name={'fullscreen-exit'}
                style={styles.fullScreenExitButtonIcon}
              />
            </Button>
          </View>
        </View>
        }
      </TouchableOpacity>
    );
  }
}


export default FullScreenVideoControls;
