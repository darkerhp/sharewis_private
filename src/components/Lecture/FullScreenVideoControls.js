import React from 'react';
import ReactNative from 'react-native';

import Button from 'react-native-button';
import Icon from 'react-native-vector-icons/MaterialIcons';

import SeekBar from './SeekBar';
import BaseStyles from '../../baseStyles';

const { PropTypes } = React;
const { Platform, StyleSheet, Text, View } = ReactNative;

const styles = StyleSheet.create({
  container: {
    flex: 2,
    justifyContent: 'space-around',
    alignItems: 'center',
    flexDirection: 'row',
  },
  playButton: {
    width: 78,
    height: 78,
    borderRadius: 78 / 2,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: BaseStyles.backgroundColor,
  },
  playButtonIcon: {
    fontSize: 40,
    height: 44,
    color: 'white',
  },
  buttonText: {
    fontSize: 18,
    color: 'white',
  },
  fullScreenButton: {
    width: 62,
    height: 62,
    borderRadius: 62 / 2,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: BaseStyles.backgroundColor,
  },
  fullScreenButtonIcon: {
    fontSize: 28,
    height: 28,
    color: 'white',
  },
});

const FullScreenVideoControls = ({
  currentTime,
  estimatedTime,
  isFullScreen,
  isLoadingThumbnail,
  isPaused,
  onPressFullScreen,
  onPressPlay,
  onPressSpeed,
  onSlidingComplete,
  onValueChange,
  speed,
  title,
}) => {
  return (
    <View
      style={{
        backgroundColor: 'transparent',
        borderRadius: 5,
        position: 'absolute',
        bottom: 20,
        left: 20,
        right: 20,
      }}
    >
      <SeekBar
        currentTime={currentTime}
        estimatedTime={estimatedTime}
        onSlidingComplete={onSlidingComplete}
        onValueChange={onValueChange}
      />
      <View style={styles.lectureTitleTextWrapper}>
        <Text style={styles.lectureTitle}>{title}</Text>
      </View>

      <View style={styles.container}>
        <Button
          containerStyle={[
            styles.playButton,
            isLoadingThumbnail && { backgroundColor: BaseStyles.disabledButtonColor },
          ]}
          style={styles.buttonText}
          onPress={() => onPressPlay()}
          disabled={isLoadingThumbnail}
        >
          <Icon
            name={isPaused ? 'play-arrow' : 'pause'}
            style={styles.playButtonIcon}
          />
        </Button>
        <Button
          containerStyle={styles.fullScreenButton}
          style={styles.buttonText}
          onPress={() => onPressFullScreen()}
        >
          <Icon
            name={'fullscreen'}
            style={styles.fullScreenButtonIcon}
          />
        </Button>
      </View>
    </View>
  );
};


FullScreenVideoControls.propTypes = {
  currentTime: PropTypes.number.isRequired,
  estimatedTime: PropTypes.number.isRequired,
  isFullScreen: PropTypes.bool.isRequired,
  isLoadingThumbnail: PropTypes.bool.isRequired,
  isPaused: PropTypes.bool.isRequired,
  speed: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  //
  onPressFullScreen: PropTypes.func.isRequired,
  onPressPlay: PropTypes.func.isRequired,
  onPressSpeed: PropTypes.func.isRequired,
  onSlidingComplete: PropTypes.func.isRequired,
  onValueChange: PropTypes.func.isRequired,
};

export default FullScreenVideoControls;
