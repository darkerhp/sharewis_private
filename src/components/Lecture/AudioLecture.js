import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ReactNative from 'react-native';

import autobind from 'autobind-decorator';
import KeepAwake from 'react-native-keep-awake';
import Orientation from 'react-native-orientation';
import Video from 'react-native-video';
import { Actions as RouterActions } from 'react-native-router-flux';
import { Client } from 'bugsnag-react-native';
import BaseStyles from '../../lib/baseStyles';

import VideoControls from './VideoControls'; // eslint-disable-line
import FullScreenVideoControls from './FullScreenVideoControls'; // eslint-disable-line
import Lecture from '../../modules/models/Lecture';

const {
  Image,
  StatusBar,
  StyleSheet,
  View,
} = ReactNative;

const styles = StyleSheet.create({
  videoContainer: {
    flex: 2,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'transparent',
    marginTop: 64,
  },
  loadingIndicatorWrapper: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
});

const lectureImageSrc = require('./images/audio_lecture_image.png');

class AudioLecture extends Component {
  static propTypes = {
    // values
    currentLecture: PropTypes.instanceOf(Lecture).isRequired,
    isFullScreen: PropTypes.bool.isRequired,
    lectureContentStyleId: PropTypes.number.isRequired,
    speed: PropTypes.number.isRequired,
    // actions
    changeVideoPlaySpeed: PropTypes.func.isRequired,
    onPlayEnd: PropTypes.func.isRequired,
    toggleFullScreen: PropTypes.func.isRequired,
  };

  static toFullScreen() {
    Orientation.lockToLandscape();
    StatusBar.setHidden(true);
    RouterActions.refresh({ hideNavBar: true });
  }

  static toPortrait() {
    Orientation.lockToPortrait();
    StatusBar.setHidden(false);
    RouterActions.refresh({ hideNavBar: false });
  }

  state = {
    seeking: false,
    isLoadingThumbnail: true,
    currentTime: 0,
    isPaused: false,
    isStarted: true,
  };

  @autobind
  handleValueChange() {
    this.setState({ seeking: true }); // 再生位置調整中はビデオを止める
  }

  @autobind
  handleSlidingComplete(value) {
    this.setState({ seeking: false });
    if (this.video) {
      this.video.seek(value);
    }
  }

  // 250ms毎に呼び出される
  @autobind
  handleVideoProgress(data) {
    if (this.state.currentTime === data.currentTime) return;
    this.setState({ currentTime: data.currentTime });
  }

  @autobind
  handlePressFullScreen() {
    const { isFullScreen, toggleFullScreen } = this.props;
    if (isFullScreen) {
      AudioLecture.toPortrait();
    } else {
      AudioLecture.toFullScreen();
    }
    toggleFullScreen();
  }

  @autobind
  handlePressPlay() {
    this.setState({ isPaused: !this.state.isPaused, isStarted: true });
  }

  render() {
    const {
      changeVideoPlaySpeed,
      currentLecture,
      isFullScreen,
      lectureContentStyleId,
      onPlayEnd,
      speed,
    } = this.props;

    const videoControlsProps = {
      currentTime: this.state.currentTime,
      estimatedTime: currentLecture.estimatedTime,
      isFullScreen,
      isLoadingThumbnail: this.state.isLoadingThumbnail,
      isPaused: this.state.isPaused,
      onPressFullScreen: this.handlePressFullScreen,
      onPressPlay: this.handlePressPlay,
      onPressSpeed: changeVideoPlaySpeed,
      onSlidingComplete: this.handleSlidingComplete,
      onValueChange: this.handleValueChange,
      speed,
      title: currentLecture.title,
    };

    return (
      <View style={[lectureContentStyleId, isFullScreen && { flex: 1 }]}>
        <View style={isFullScreen ? { flex: 1 } : styles.videoContainer}>
          <Image
            style={
              isFullScreen
                ? { height: BaseStyles.deviceWidth, width: BaseStyles.deviceHeight }
                : { width: BaseStyles.deviceWidth }
            }
            source={lectureImageSrc}
            resizeMode="contain"
          />
          <Video
            muted={false}
            onLoadStart={() => this.setState({ isLoadingThumbnail: true })}
            onLoad={() => this.setState({ isLoadingThumbnail: false })}
            onEnd={onPlayEnd}
            onError={(e) => {
              new Client().notify(e);
              console.error(e);
            }}
            onProgress={this.handleVideoProgress}
            paused={this.state.seeking || this.state.isPaused}
            playInBackground
            playWhenInactive={false}
            rate={speed}
            ref={(ref) => { this.video = ref; }}
            repeat={false}
            resizeMode="contain"
            source={{ uri: currentLecture.getAudioUrl() }}
            volume={1.0}
          />
        </View>

        {isFullScreen
          ? <FullScreenVideoControls {...videoControlsProps} />
          : <VideoControls {...videoControlsProps} />}

        <KeepAwake />
      </View>
    );
  }
}

export default AudioLecture;
