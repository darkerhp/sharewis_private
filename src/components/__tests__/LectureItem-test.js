/* eslint-disable no-undef */
/* eslint-disable import/no-extraneous-dependencies */
import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import LectureItem from '../CourseDetails/LectureItem';

import Lecture from '../../models/Lecture';

jest.mock('../Duration', () => 'Duration');
jest.mock('bugsnag-react-native', () => 'Bugsnag');

describe('LectureItem', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <LectureItem
        isOnline
        handlePressDelete={() => null}
        handlePressDownload={() => null}
        handlePressLecture={() => null}
        lecture={new Lecture({
          status: 'not_started',
          order: 1,
          estimatedTime: 30,
          title: 'Title',
          type: 'video',
        })}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
