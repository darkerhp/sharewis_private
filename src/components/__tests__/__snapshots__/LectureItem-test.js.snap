exports[`LectureItem renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "borderBottomWidth": 1,
      "borderColor": "#ddd",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "flex-start",
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#ddd",
        "borderRightWidth": 1,
        "flex": 1,
        "height": 48,
        "justifyContent": "center",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#525252",
            "fontSize": 14,
            "fontWeight": "600",
          },
          Object {},
        ]
      }>
      1
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "height": 48,
          "justifyContent": "center",
        },
        Object {},
      ]
    }>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#525252",
              "fontSize": 15,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }>
        
      </Text>
    </View>
    <Duration
      containerStyleId={
        Object {
          "flex": 1,
        }
      }
      durationStyleId={
        Object {
          "backgroundColor": "#f2f2f2",
          "fontSize": 8,
          "fontWeight": "bold",
          "padding": 3,
          "textAlign": "center",
        }
      }
      estimatedTime={30} />
  </View>
  <TouchableOpacity
    activeOpacity={0.2}
    disabled={false}
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "flex": 5,
          "height": 48,
          "justifyContent": "center",
        },
        Object {},
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#525252",
            "fontSize": 12,
            "fontWeight": "bold",
          },
          Object {},
        ]
      }>
      Title
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": 48,
        "justifyContent": "center",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "#757575",
            "fontSize": 32,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }>
      
    </Text>
  </TouchableOpacity>
</View>
`;
